### DISCUSSION TITLE
Feature Request: Use description frontmatter from single page in steps shortcode.

### DISCUSSION THREAD

**Discussion by ghost** [imfing/hextra#36] [Ideas]
Use case:

Let's say there's a list page like [this](https://imfing.github.io/hextra/docs/guide/shortcodes/steps/) and "Step 1", "Step 2", and "Step 3" are individual pages.

Under each individual page, there's text "This is the first step.", "This is the second step.", etc.

It would be very useful if "This is the first step." can be retrieved from the `description` front matter of the single page `step-1.md`.

If possible, please include a condition in the "steps" shortcode, to retrieve the description from the single page, if its available.

**Comment by imfing** [comment]
Gotcha, I think this can be approached in this way. 

Say we include a shortcode `layouts/shortcodes/readFile.html` ([example](https://github.com/gohugoio/hugo/blob/master/docs/layouts/shortcodes/readfile.html)):

```
{{% steps %}}

{{% readFile "step1.md" %}}

{{% readFile "step2.md" %}}

{{% readFile "step3.md" %}}

{{% /steps %}}
```

which in effect will be equivalent to:

```
{{% steps %}}

### Step 1

This is the first step.

### Step 2

This is the second step.

### Step 3

This is the third step.

{{% /steps %}}
```

and if you only want to read the description from the frontmatter, then you can create a similar shortcode `layouts/shortcodes/readDescription.html` which does the similar thing

**Reply by ghost** [reply]
That's a good idea! I'll work on the `readDescription.html` shortcode and submit a PR soon.

In my opinion, a shortcode like `readFile.html` may be too verbose as its reading the entire file, and may not fit well with Hextra as it's a documentation theme.

The `readDescription` shortcode will fit in very well with the `steps`. The use case I have in mind is  if sections are big enough to contain a dedicated document (like `step1.md`, `step2.md`), and there are many such sections, then pulling in the description will be useful.

**Reply by imfing** [reply]
Thanks for the proposal and I appreciate your thought.
While the `readDescription.html` shortcode would work for the steps, it seems to address a very specific use case. 
I'm planning to focus more on broader functionality such as `readFile`, which I believe will have more general applicability.

However, feel free to share the `readDescription.html` you create here, as it could be useful for others who have similar requirements.

**Comment by ghost** [comment]
You're right - the use case for `readDescription.html` is very limited.
